pkg/cleaner/cleaner.go

```diff
@@ -0,0 +1,90 @@
+// Package cleaner is a background process that compares the kubernetes namespace list with the folders in the local git home directory, deleting what's not in the namespace list
+package cleaner
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/deis/builder/pkg/k8s"
+	"github.com/deis/builder/pkg/sshd"
+	"github.com/deis/pkg/log"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+func localDirs(gitHome string) ([]string, error) {
+	var ret []string
+	err := filepath.Walk(gitHome, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return filepath.SkipDir
+		}
+		ret = append(ret, filepath.Join(gitHome, path))
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+// getDiff gets the directories that are not in namespaceList
+func getDiff(namespaceList []api.Namespace, dirs []string) []string {
+	var ret []string
+
+	// create a set of lowercase namespace names
+	namespacesSet := make(map[string]struct{})
+	for _, ns := range namespaceList {
+		lowerName := strings.ToLower(ns.Name)
+		namespacesSet[lowerName] = struct{}{}
```

Comment:
Tracked down, `[a-z0-9-]+` is the Application Regex, so that's the namespace regex as well